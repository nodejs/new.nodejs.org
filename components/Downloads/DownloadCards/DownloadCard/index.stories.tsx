import DownloadCard from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof DownloadCard>;
type Meta = MetaObj<typeof DownloadCard>;

export const Selected: Story = {
  args: {
    os: 'MAC',
    icon: 'mac-download-logo.svg',
    label: 'MAC Installer',
    download: 'https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg',
    filename: 'node-v18.15.0.pkg',
    selected: true,
    onSelect: () => {},
  },
};

export const NotSelected: Story = {
  args: {
    os: 'MAC',
    icon: 'mac-download-logo.svg',
    label: 'MAC Installer',
    download: 'https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg',
    filename: 'node-v18.15.0.pkg',
    selected: false,
    onSelect: () => {},
  },
};

export default { component: DownloadCard } as Meta;
