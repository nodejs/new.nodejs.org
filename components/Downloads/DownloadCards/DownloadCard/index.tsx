import classNames from 'classnames';
import { MdGetApp } from 'react-icons/md';
import { Tab } from 'react-tabs';
import Image from 'next/image';
import styles from './index.module.scss';
import { useRouter } from '@/hooks/useRouter';
import type { FC } from 'react';
import type { UserOS } from '@/types/userOS';

type DownloadCardProps = {
  os: UserOS;
  icon: string;
  label: string;
  download: string;
  filename: string;
  selected: boolean;
  onSelect: (os: string) => void;
};

const DownloadCard: FC<DownloadCardProps> = ({
  os,
  icon,
  label,
  download,
  filename,
  selected,
  onSelect,
}) => {
  const { basePath } = useRouter();

  const tabClassNames = classNames(styles.downloadCard, {
    [styles.downloadCardActive]: selected,
  });
  const selectCard = () => onSelect(os);
  const size = selected ? 56 : 48;

  return (
    <Tab
      className={tabClassNames}
      id={`download-card-${os}`}
      key={os}
      onClick={selectCard}
      selectedClassName={styles.downloadCardActive}
      tabIndex="0"
    >
      <div className={styles.top}>
        <Image
          height={size}
          width={size}
          src={`${basePath}/static/images/logos/${icon}`}
          alt={label}
        />
        {selected && (
          <a className={styles.link} href={download}>
            <MdGetApp />
          </a>
        )}
      </div>
      <p className={styles.label}>{label}</p>
      <p className={styles.filename} data-testid="lts">
        {filename}
      </p>
    </Tab>
  );
};

export default DownloadCard;
