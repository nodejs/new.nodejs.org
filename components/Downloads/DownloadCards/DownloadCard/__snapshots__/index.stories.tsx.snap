// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downloads/DownloadCards/DownloadCard NotSelected smoke-test 1`] = `
<li role="tab"
    id="tabdownload-card-MAC"
    aria-selected="false"
    aria-disabled="false"
    aria-controls="paneldownload-card-MAC"
    tabindex="0"
    data-rttab="true"
>
  <div>
    <img alt="MAC Installer"
         loading="lazy"
         width="48"
         height="48"
         decoding="async"
         data-nimg="1"
         srcset="/static/images/logos/mac-download-logo.svg?w=48&amp;q=75 1x, /static/images/logos/mac-download-logo.svg?w=96&amp;q=75 2x"
         src="/static/images/logos/mac-download-logo.svg?w=96&amp;q=75"
         style="color: transparent;"
    >
  </div>
  <p>
    MAC Installer
  </p>
  <p data-testid="lts">
    node-v18.15.0.pkg
  </p>
</li>
`;

exports[`Downloads/DownloadCards/DownloadCard Selected smoke-test 1`] = `
<li role="tab"
    id="tabdownload-card-MAC"
    aria-selected="false"
    aria-disabled="false"
    aria-controls="paneldownload-card-MAC"
    tabindex="0"
    data-rttab="true"
>
  <div>
    <img alt="MAC Installer"
         loading="lazy"
         width="56"
         height="56"
         decoding="async"
         data-nimg="1"
         srcset="/static/images/logos/mac-download-logo.svg?w=64&amp;q=75 1x, /static/images/logos/mac-download-logo.svg?w=128&amp;q=75 2x"
         src="/static/images/logos/mac-download-logo.svg?w=128&amp;q=75"
         style="color: transparent;"
    >
    <a href="https://nodejs.org/dist/v18.15.0/node-v18.15.0.pkg">
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 24 24"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
      >
        <path fill="none"
              d="M0 0h24v24H0z"
        >
        </path>
        <path d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z">
        </path>
      </svg>
    </a>
  </div>
  <p>
    MAC Installer
  </p>
  <p data-testid="lts">
    node-v18.15.0.pkg
  </p>
</li>
`;
